File: backend/modules/airspace-module/airspace-domain-module/src/main/java/pl/drone_zone/api/models/Airspace.java
package pl.drone_zone.api.models;

import lombok.Builder;

@Builder(toBuilder = true)
public record Airspace(
        String id,
        String name,
        String color,
        int type,
        int icaoClass,
        boolean onDemand,
        boolean onRequest,
        boolean byNotam,
        boolean specialAgreement,
        boolean requestCompliance,
        HeightLimit upperLimit,
        HeightLimit lowerLimit,
        Geometry geometry
) {
}

File: frontend/src/models/airspace.ts
interface Coordinate {
  lat: number;
  lng: number;
}

export interface Geometry {
  type: string;
  coordinates: Coordinate[][];
}

export interface HeightLimit {
  value: number;
  unit: number;
}

export enum Category {
  OPEN,
}

export enum SubCategory {
  A1 = "A1",
  A2 = "A2",
  A3 = "A3",
}
export interface Airspace {
  id: string;
  name: string;
  color: string;
  type: number;
  icaoClass: number;
  activity: number;
  onDemand: boolean;
  onRequest: boolean;
  byNotam: boolean;
  specialAgreement: boolean;
  requestCompliance: boolean;
  upperLimit: HeightLimit;
  lowerLimit: HeightLimit;
  geometry: Geometry;
}
